This fixes a bug where HELPFILE was being overridden by a hard coded path

This also fixes an issue with fl_filename_absolute returning a relative path
preventing fteapot from finding its helpfile when started from any directory
except /

Patch by Kevin McCarthy <signals42@gmail.com>

--- a/fteapot.fl
+++ b/fteapot.fl
@@ -5,6 +5,10 @@
 decl {\#include <stdint.h>} {private global
 }
 
+decl {\#include "config.h"} {private global
+}
+
+
 decl {\#include <limits.h>} {private global
 }
 
@@ -824,9 +828,9 @@
 
 Function {find_helpfile(char *buf, int size, const char *argv0)} {open C return_type void
 } {
-  code {fl_filename_absolute(buf, size, argv0);
+  code {strncpy(buf, size, argv0);
 char *p = (char *)fl_filename_name(buf);
-strncpy(p, "../share/doc/teapot/html/index.html", buf+size-p);
+strncpy(p, HELPFILE, buf+size-p);
 buf[size-1] = 0;
 
 // Check if help exists in default installed location, fallback value is valid for build directory
